usersession/agent/session_agent_test.go

```diff
@@ -151,6 +155,93 @@ func (s *sessionAgentSuite) TestExitOnIdle(c *C) {
 	}
 }
 
+func (s *sessionAgentSuite) TestFdoNotification(c *C) {
+	desktopFile := "[Desktop Entry]\nIcon=/path/appicon.png"
+	c.Assert(os.MkdirAll(filepath.Join(dirs.SnapDesktopFilesDir), 0755), IsNil)
+	c.Assert(ioutil.WriteFile(filepath.Join(dirs.SnapDesktopFilesDir, "app.desktop"), []byte(desktopFile), 0644), IsNil)
+
+	backend, err := notificationtest.NewFdoServer()
+	c.Assert(err, IsNil)
+	defer backend.Stop()
+
+	agent, err := agent.New()
+	c.Assert(err, IsNil)
+	agent.IdleTimeout = 150 * time.Millisecond
+	agent.Start()
+	defer agent.Stop()
+
+	// simulate snap refresh message
+	makeRequest := func() {
+		data := bytes.NewBufferString(`{"instance-name":"some-snap", "busy-app-name":"App", "busy-app-desktop-entry":"app"}`)
+		response, err := s.client.Post("http://localhost/v1/notifications/pending-refresh", "application/json", data)
+		c.Assert(err, IsNil)
+		defer response.Body.Close()
+		c.Check(response.StatusCode, Equals, 200)
+	}
+	makeRequest()
+
+	// wait for a while, we want the message to trigger FDO notification
+	time.Sleep(50 * time.Millisecond)
```

Comment:
\<insert obligatory comment about challenging the deities of slowness that are LP builders\>

Reply:
Fair point; I have a few aggressive timeouts/delays in this test suite, not sure it makes sense to add comments to all of them, I'd say let's see how it goes and maybe they will need to be scaled up.